lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex

```
@@ -55,18 +55,19 @@ defmodule LambdaEthereumConsensus.ForkChoice do
         :telemetry.execute([:sync, :on_block], %{slot: slot})
         Logger.info("[Fork choice] Added new block", slot: slot, root: block_root)
 
-        Task.start_link(__MODULE__, :recompute_head, [new_store])
+        :telemetry.span([:fork_choice, :recompute_head], %{}, fn ->
+          {recompute_head(new_store), %{}}
+        end)
 
         %Store{finalized_checkpoint: new_finalized_checkpoint} = new_store
 
         prune_old_states(last_finalized_checkpoint.epoch, new_finalized_checkpoint.epoch)
 
         persist_store(new_store)
-        GenServer.cast(from, {:block_processed, block_root, true})
 
       {:error, reason} ->
         Logger.error("[Fork choice] Failed to add block: #{reason}", slot: slot, root: block_root)
-        GenServer.cast(from, {:block_processed, block_root, false})
+        :error
```

We should return the error with its reason and let the caller handle it.